rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -581,9 +581,21 @@ private class BlockExprItemNode extends ItemNode instanceof BlockExpr {
 }
 
 class TypeParamItemNode extends ItemNode instanceof TypeParam {
+  private WherePred getAWherePred() {
+    exists(ItemNode declaringItem |
```

Comment:
The type of `declaringItem` is not very descriptive. Any item with `result` and `this` as descendants would fit. Could we reasonably spell out the type of items we expect here or does this cover to many things that that would be too tedious?

Reply:
I think it would be a bit tedious, type parameters can be declared in many different places: functions, type aliases, structs, enums, unions, consts, traits, impl blocks.